File: day01/Star1.java
import static java.lang.Integer.parseInt;
import static java.lang.System.nanoTime;
import static java.lang.System.out;
import static java.util.function.Predicate.not;

void main() throws Exception {
    doFile("input-test1.txt");
    doFile("input.txt");
    out.printf("*** %s *** DONE ***%n", LocalTime.now());
}

static final Pattern SEPARATOR = Pattern.compile(" {3}");

long fileStartTime = 0;
boolean test = false;

void doFile(String filename) throws Exception {
    out.printf("*** %s *** input file: %s ***%n", LocalTime.now(), filename);
    test = filename.contains("test");
    fileStartTime = nanoTime();

    var uri = getClass().getResource(filename)
            .toURI();
    try (var lines = Files.lines(Paths.get(uri))) {
        var input = lines.filter(not(String::isBlank))
                .map(String::trim)
                .map(SEPARATOR::split)
                .map(xs -> new int[]{parseInt(xs[0]), parseInt(xs[1])})
                .toList();

        int[] list1 = input.stream().mapToInt(xs -> xs[0]).sorted().toArray();
        int[] list2 = input.stream().mapToInt(xs -> xs[1]).sorted().toArray();


        int solution = 0;
        for (int i = 0; i < list1.length; i++)
            solution += Math.abs(list1[i] - list2[i]);

        out.println("solution = " + solution);
        out.printf("%,5d ns%n", nanoTime() - fileStartTime);
    }

    out.println();
}

File: day01/star1.js
import {readFileSync} from 'node:fs';

let test;

const doFile = (filename) => {
    console.group(`*** ${new Date().toLocaleTimeString()} *** input file: ${filename} ***`);
    console.time(filename);
    test = filename.includes('test');

    const lines = readFileSync(filename, 'utf8')
        .split('\n')
        .map(l => l.trim())
        .filter(l => l);

    const input = lines.map(l => l.split('   '));
    const list1 = input.map(xs => +xs[0]).sort((a, b) => a - b);
    const list2 = input.map(xs => +xs[1]).sort((a, b) => a - b);

    let solution = 0;
    for (let i = 0; i < list1.length; i++)
        solution += Math.abs(list1[i] - list2[i]);

    console.timeEnd(filename);
    console.groupEnd();
    console.log(`* Solution: ${solution}\n`);
};

(() => {
    doFile('input-test1.txt');
    doFile('input.txt');
    console.log(`*** ${new Date().toLocaleTimeString()} *** DONE`);
})();
